app/components/add-layer/component.js

```diff
@@ -152,14 +154,53 @@ export default Ember.Component.extend({
                     }
                 };
                 break;
+            }
+            let index = this.get('index')+1;
+            if(this.get('indexVal') !== null){
+                index = parseInt(this.get('indexVal'))+1 
+            }
+            this.get('layers.content').insertAt(index,item);
+            this.set('isOpen', false);
+            this.set('indexVal' , null)
+
+        },
+        toggleProperty(prop){
+            this.toggleProperty(prop);
+        },
+        allowDragOver(event){
+            $('.add-layer-toggle').addClass('dotted-line');
+            // $(event.target.parentNode).closest('.add-layer-toggle').addClass('dotted-line') //save 
+            $('.drag-drop-area').css('padding' , '5px');
+            $('.drop-zone-plus').css('display' , 'block');
+            $('.plus').css('display' , 'none');
```

Comment:
the jquery on these handlers could be replaced by binding the classes/styles through the template